include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -928,10 +1023,9 @@ __parallel_scan_copy(oneapi::dpl::__internal::__device_backend_tag __backend_tag
 
     return __parallel_transform_scan_base(
         __backend_tag, ::std::forward<_ExecutionPolicy>(__exec),
-        oneapi::dpl::__ranges::make_zip_view(
-            ::std::forward<_InRng>(__in_rng),
-            oneapi::dpl::__ranges::all_view<int32_t, __par_backend_hetero::access_mode::read_write>(
-                __mask_buf.get_buffer())),
+        oneapi::dpl::__ranges::zip_view(
+            __in_rng, oneapi::dpl::__ranges::all_view<int32_t, __par_backend_hetero::access_mode::read_write>(
+                          __mask_buf.get_buffer())),
```

Comment:
I don't like that this change is required, but it mirrors how it is currently handled in the ranges implementation. https://github.com/oneapi-src/oneDPL/blob/aec12a4c55ac5de94753f8e3259b50cd6befb0c3/include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h#L364
Its needed here because we are unifying the ranges API and main API handling of `__parallel_copy_if` to enable the performance improvements to be felt in both cases, and also to remove redundant code.

This is to avoid an issue with making a `zip_view` from a non const lvalue reference to a zip view.  There may be changes we can make to our internal tuple class to resolve this, but I think that lies outside the scope of this PR.

Reply:
Is there a GitHub issue for this? It may be good to create one if there is not.

Reply:
nope, I'll make one.